File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/authorization/SemanticAuthorizationResult.java
package com.foundgine.core.semantic.authorization;
import com.foundgine.core.semantic.ir.SemanticOperation; import com.foundgine.core.semantic.SemanticContractSnapshot; import java.util.*;
public record SemanticAuthorizationResult(SemanticOperation operation,SemanticAuthorizationEvidence evidence){public SemanticAuthorizationResult{Objects.requireNonNull(operation);Objects.requireNonNull(evidence);}public void ensureMatches(SemanticContractSnapshot c){evidence.ensureMatches(c);}}

File: src/csharp/Foundgine.Core/Semantic/Authorization/SemanticAuthorizationResult.cs
using Foundgine.Core.Semantic.IR;

namespace Foundgine.Core.Semantic.Authorization;

/// <summary>Authorization output that cannot be detached from its contract identity.</summary>
public sealed record SemanticAuthorizationResult(
    SemanticOperation Operation,
    SemanticAuthorizationEvidence Evidence)
{
    public void EnsureMatches(SemanticContractSnapshot contract)
    {
        ArgumentNullException.ThrowIfNull(contract);
        Evidence.EnsureMatches(contract);
    }
}